src/middleware/findOwner.js

```diff
@@ -17,7 +17,7 @@ const isOwner = async (req, res, next) => {
     });
   }
   const isAuthor = foundArticle.authorId === userId;
-  if ((!isAuthor) && (role !== ('admin' || 'super-admin'))) {
+  if ((!isAuthor) && ((role !== 'admin') && (role !== 'super-admin'))) {
```

Comment:
shouldn't this be `(role !== 'admin') || (role !== 'super-admin')`

Reply:
@mkiterian We tried that and didn't work because the roles are grouped in the same bracket. That one could work if the roles are not grouped in the same bracket. if we use the ```||``` operator because they are grouped together if one of them is false the whole tautology becomes false and the message will not be executed.

Reply:
what do you mean by grouped in the same bracket?

Reply:
@mkiterian  ``` ((!isAuthor) && ((role !== 'admin') && (role !== 'super-admin'))) ```
after the first && there is a bracket that is grouping (role !== admin) and (role !== super-admin) together and if we try to ungroup them the eslint won't allow it to mix two different operators that is why we grouped them in the same bracket

Reply:
@mkiterian the logic there was not working as intended because when the second part of the tautology returns a True value and the user is not the author then its resulting value becomes True. So we had to refactor it a bit.